source/MaterialXCore/Document.cpp

```diff
@@ -1061,6 +1061,48 @@ void Document::upgradeVersion(bool applyFutureUpdates)
                     }
                 }
             }
+
+            // Make it so that interface names and nodes in a nodegraph are not duplicates
+            // If they are, rename the nodes.
+            std::set<string> interfaceNames;
+            for (NodeGraphPtr nodegraph : getNodeGraphs())
+            {
+                for (ElementPtr child : nodegraph->getChildren())
+                {
+                    NodePtr srcNode = child->asA<Node>();
+                    if (srcNode)
+                    {
+                        for (ValueElementPtr elem : srcNode->getChildrenOfType<ValueElement>())
+                        {
+                            const string& interfaceName = elem->getInterfaceName();
+                            if (!interfaceName.empty())
+                            {
+                                interfaceNames.insert(interfaceName);
+                            }
+                        }
+                    }
+                }
+            }
+            for (string interfaceName : interfaceNames)
+            {
+                for (NodeGraphPtr nodegraph : getNodeGraphs())
```

Comment:
This is not correct. You don't need uniqueness across all nodegraph nodes in the document.